pkg/api/v1/resource/helpers.go

```diff
@@ -99,7 +106,7 @@ func PodRequests(pod *v1.Pod, opts PodResourcesOptions) v1.ResourceList {
 			containerReqs = applyNonMissing(containerReqs, opts.NonMissingContainerRequests)
 		}
 
-		if container.RestartPolicy != nil && *container.RestartPolicy == v1.ContainerRestartPolicyAlways {
+		if isSidecar(&container) {
```

Comment:
SidecarContainers feature is currently in beta. Do we need to consider feature gates when making this change?

Reply:
For unscheduled pods, scheduler will reject it first if `SidecarContainers` is not enabled for scheduler and the pod has one.
For a scheduled and running pod, I think we should always take it into consideration.